tests/testlib/s2n_test_server_client.c

```diff
@@ -77,8 +77,15 @@ S2N_RESULT s2n_negotiate_test_server_and_client_with_early_data(struct s2n_conne
     bool server_early_done = false, client_early_done = false;
     do {
         if (!client_early_done) {
+            /* Adding 0 to a null value is undefined behavior.
+            * This prevents undefined behavior in the case where the offset is 0.
+            */
+            uint8_t *early_data = early_data_to_send->data;
+            if (total_data_sent != 0) {
+                early_data += total_data_sent;
+            }
```

Comment:
Nit: You turned "early_data_to_send" into a more general variable instead of a more specific variable
```suggestion
            /* Adding 0 to a null value is undefined behavior.
             * This prevents undefined behavior in the case where the offset is 0.
             */
            uint8_t *next_early_data_to_send = early_data_to_send->data;
            if (total_data_sent != 0) {
                next_early_data_to_send += total_data_sent;
            }
```